.devcontainer/Dockerfile

```diff
@@ -0,0 +1,39 @@
+ARG VARIANT=12
```

Comment:
I'll separate out the .devcontainer changes and commit separately.  This one still needs a bit of work.

If you're not familiar: https://code.visualstudio.com/docs/remote/containers

Easiest way to try it out:
* Install WSL2
* Install Docker Desktop
* Install "Remote: Containers" extension
* Use "> Remote-Containers: Open repository in remote container..."

Viola...  preconfigured dev environment that is 100% isolated from your dev machine, but still "feels local".